File: SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/Servicio/ServicioDTO.java
package pe.edu.pucp.softinv.model.Servicio;

import pe.edu.pucp.softinv.model.Comentario.ComentarioServicioDTO;
import pe.edu.pucp.softinv.model.Personas.EmpleadoDTO;

import java.util.ArrayList;

public class ServicioDTO {
    private Integer idServicio;
    private String nombre;
    private String descripcion;
    private TipoServicio tipo;
    private Double precio;
    private Double promedioValoracion;
    private String urlImagen;
    private ArrayList<ComentarioServicioDTO> comentarios;
    private Integer duracionHora;
    private ArrayList<EmpleadoDTO> empleados;
    private ArrayList<CitaDTO> citas;

    public Integer getIdServicio() {
        return idServicio;
    }

    public void setIdServicio(Integer idServicio) {
        this.idServicio = idServicio;
    }

    public ArrayList<CitaDTO> getCitas() {
        return citas;
    }

    public void setCitas(ArrayList<CitaDTO> citas) {
        this.citas = citas;
    }

    public ArrayList<EmpleadoDTO> getEmpleados() {
        return empleados;
    }

    public void setEmpleados(ArrayList<EmpleadoDTO> empleados) {
        this.empleados = empleados;
    }

    public Integer getDuracionHora() {
        return duracionHora;
    }

    public void setDuracionHora(Integer duracionMin) {
        this.duracionHora = duracionMin;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getTipo() {
        if(tipo==TipoServicio.FACIAL)
            return "FACIAL";
        else if(tipo==TipoServicio.CORPORAL)
            return "CORPORAL";
        else if(tipo==TipoServicio.TERAPIA_COMPLEMENTARIA)
            return "TERAPIA_COMPLEMENTARIA";
        return "";
    }

    public void setTipo(TipoServicio tipo) {
        this.tipo = tipo;
    }

    public void setTipo(String tipo) {
        if(tipo == "FACIAL")
            this.tipo = TipoServicio.FACIAL;
        else if (tipo == "CORPORAL")
            this.tipo = TipoServicio.CORPORAL;
        else if (tipo == "TERAPIA_COMPLEMENTARIA")
            this.tipo = TipoServicio.TERAPIA_COMPLEMENTARIA;
    }

    public Double getPrecio() {
        return precio;
    }

    public void setPrecio(Double precio) {
        this.precio = precio;
    }

    public Double getPromedioValoracion() {
        return promedioValoracion;
    }

    public void setPromedioValoracion(Double promedioValoracion) {
        this.promedioValoracion = promedioValoracion;
    }

    public String getUrlImagen() {
        return urlImagen;
    }

    public void setUrlImagen(String urlImagen) {
        this.urlImagen = urlImagen;
    }

    public ArrayList<ComentarioServicioDTO> getComentarios() {
        return comentarios;
    }

    public void setComentarios(ArrayList<ComentarioServicioDTO> comentarios) {
        this.comentarios = comentarios;
    }

    public ServicioDTO() {
        idServicio=null;
        nombre = null;
        tipo = null;
        precio = null;
        promedioValoracion = null;
        urlImagen = null;
        comentarios = null;
        duracionHora=null;
        empleados=null;
        citas=null;
        descripcion = null;
    }

    public ServicioDTO(Integer idServicio,String nombre,TipoServicio tipo,Double precio,Double promedioValoracion,
                       String urlImagen,ArrayList<ComentarioServicioDTO> comentarios, Integer duracionHora,
                       ArrayList<EmpleadoDTO> empleados,ArrayList<CitaDTO> citas, String descripcion) {
        this.idServicio=idServicio;
        this.nombre=nombre;
        this.tipo=tipo;
        this.precio=precio;
        this.promedioValoracion=promedioValoracion;
        this.urlImagen=urlImagen;
        this.comentarios=comentarios;
        this.duracionHora=duracionHora;
        this.empleados=empleados;
        this.citas=citas;
        this.descripcion = descripcion;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
}

File: C#/SoftInvModel/Servicio/ServicioDTO.cs
using System;
using System.ComponentModel;

namespace softinv.model
{
    public class ServicioDTO
    {
        private int? idServicio;
        private string nombre;
        private string descripcion;
        private TipoServicio tipo;
        private double? precio;
        private double? promedioValoracion;
        private string urlImagen;
        private BindingList<ComentarioServicioDTO> comentarios;
        private int? duracionHora;
        private BindingList<EmpleadoDTO> empleados;
        private BindingList<CitaDTO> citas;

        public int? IdServicio { get => idServicio; set => idServicio = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }
        public TipoServicio Tipo { get => tipo; set => tipo = value; }
        public double? Precio { get => precio; set => precio = value; }
        public double? PromedioValoracion { get => promedioValoracion; set => promedioValoracion = value; }
        public string UrlImagen { get => urlImagen; set => urlImagen = value; }
        public BindingList<ComentarioServicioDTO> Comentarios { get => comentarios; set => comentarios = value; }
        public int? DuracionHora { get => duracionHora; set => duracionHora = value; }
        public BindingList<EmpleadoDTO> Empleados { get => empleados; set => empleados = value; }
        public BindingList<CitaDTO> Citas { get => citas; set => citas = value; }

        public string getTipo()
        {
            if (Tipo == TipoServicio.FACIAL)
                return "FACIAL";
            else if (Tipo == TipoServicio.CORPORAL)
                return "CORPORAL";
            else if (Tipo == TipoServicio.TERAPIA_COMPLEMENTARIA)
                return "TERAPIA_COMPLEMENTARIA";
            return "";
        }

        public void setTipo(string tipo)
        {
            if (tipo == "FACIAL")
                this.Tipo = TipoServicio.FACIAL;
            else if (tipo == "CORPORAL")
                this.Tipo = TipoServicio.CORPORAL;
            else if (tipo == "TERAPIA_COMPLEMENTARIA")
                this.Tipo = TipoServicio.TERAPIA_COMPLEMENTARIA;
        }

        public ServicioDTO()
        {
            this.IdServicio = null;
            this.Nombre = null;
            this.Precio = null;
            this.PromedioValoracion = null;
            this.UrlImagen = null;
            this.Comentarios = null;
            this.DuracionHora = null;
            this.Empleados = null;
            this.Citas = null;
            this.Descripcion = null;
        }

        public ServicioDTO(int idServicio, string nombre, TipoServicio tipo, double precio, double promedioValoracion,
                           string urlImagen, BindingList<ComentarioServicioDTO> comentarios, int duracionHora,
                           BindingList<EmpleadoDTO> empleados, BindingList<CitaDTO> citas, string descripcion)
        {
            this.IdServicio = idServicio;
            this.Nombre = nombre;
            this.Tipo = tipo;
            this.Precio = precio;
            this.PromedioValoracion = promedioValoracion;
            this.UrlImagen = urlImagen;
            this.Comentarios = comentarios;
            this.DuracionHora = duracionHora;
            this.Empleados = empleados;
            this.Citas = citas;
            this.Descripcion = descripcion;
        }
    }

}